src/VisualStudio/TestUtilities/IntegrationHost.cs

```diff
@@ -0,0 +1,522 @@
+
+                var displayNameProcessId = 0;
+
+                if (!int.TryParse(fullDisplayName.Split(':').Last(), out displayNameProcessId))
+                {
+                    continue;
+                }
+
+                var displayName = fullDisplayName.Substring(0, (fullDisplayName.Length - (displayNameProcessId.ToString().Length + 1)));
+                var fullProgId = VsProgId.StartsWith("!") ? VsProgId : $"!{VsProgId}";
+
+                if (displayName.Equals(fullProgId, StringComparison.OrdinalIgnoreCase))
+                {
+                    if (displayNameProcessId == process.Id)
+                    {
+
+                        runningObjectTable.GetObject(moniker, out dte);
+                    }
+                }
+            }
+            while (dte == null);
+
+            return (DTE)(dte);
+        }
+
+        public void Cleanup()
+        {
+            // Clean the Backup Files folder so we avoid the recovery dialog on launch.
+            IntegrationHelper.DeleteDirectoryRecursively(Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.MyDocuments), VsUserFilesFolder, "Backup Files"));
+
+            CleanupDte();
+
+            if (RequireNewInstance)
+            {
+                IntegrationLog.Current.WriteLine("Closing existing Visual Studio instance.");
+
+                CleanupTimeoutWatcherThread();
+                CleanupRemotingService();
+                CleanupModalWatcherThread();
+                CleanupHostProcess();
+            }
+        }
+
+        public void ClickAutomationElement(string elementName, bool recursive = false, int timeout = IntegrationHelper.DefaultTimeout)
+        {
+            var automationElement = LocateAutomationElement(elementName, recursive, timeout);
+
+            var invokePattern = null as object;
+            if (automationElement.TryGetCurrentPattern(InvokePattern.Pattern, out invokePattern))
+            {
+                ((InvokePattern)(invokePattern)).Invoke();
+            }
+        }
+
+        public void Dispose()
+        {
+            Dispose(true);
+            // We explicitly don't call GC.SuppressFinalize() so that
+            // the finalizer is still called and the _hostProcess is killed
+        }
+
+        public void Initialize()
+        {
+            Cleanup();
+
+            _timeoutStart = IntegrationHelper.CurrentTickCount;
+
+            if (RequireNewInstance)
+            {
+                IntegrationLog.Current.WriteLine("Starting a new instance of Visual Studio.");
+
+                InitializeHostProcess();
+                InitializeDte();
+                InitializeRemotingService();
+                InitializeTimeoutWatcherThread();
+                InitializeModalWatcherThread();
+            }
+
+            // TODO: Close Interactive Window
+            // TODO: Reset Roslyn Options
+
+            // TODO: Enable Tracking of Active Token
+            // TODO: Disable Partial Solutions
```

Comment:
Please file bugs for the TODO and reference them here or remove the TODO if they are no longer necessary.